test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,526 @@
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+TEST(TransformationChangingMemorySemanticsTest, NotApplicable) {
+  const std::string shader = R"(
+               OpCapability Shader
+               OpCapability Int8
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 320
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1
+          %7 = OpTypeInt 8 1
+          %9 = OpTypeInt 32 0
+         %26 = OpTypeFloat 32
+          %8 = OpTypeStruct %6
+         %10 = OpTypePointer StorageBuffer %8
+         %11 = OpVariable %10 StorageBuffer
+         %19 = OpConstant %26 0
+         %18 = OpConstant %9 1
+         %12 = OpConstant %6 0
+         %13 = OpTypePointer StorageBuffer %6
+         %15 = OpConstant %6 4
+         %16 = OpConstant %6 7
+         %17 = OpConstant %7 4
+         %20 = OpConstant %6 80 ; SequentiallyConsistent | UniformMemory
+         %97 = OpConstant %6 64 ; None | UniformMemory
+         %98 = OpConstant %6 66 ; Acquire | UniformMemory
+         %99 = OpConstant %6 68 ; Release | UniformMemory
+        %100 = OpConstant %6 72 ; AcquireRelease | UniformMemory
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %14 = OpAccessChain %13 %11 %12
+         %21 = OpAtomicLoad %6 %14 %15 %20
+         %23 = OpAtomicExchange %6 %14 %15 %100 %16
+         %25 = OpAtomicCompareExchange %6 %14 %15 %20 %97 %16 %15
+         %24 = OpAccessChain %13 %11 %12
+               OpAtomicStore %14 %15 %99 %16
+               OpReturn
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Bad: Instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(150, SpvOpAtomicLoad, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+#ifndef NDEBUG
+
+  // Bad: Operand position does not equal 0 or 1.
+  ASSERT_DEATH(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 2, 100)
+                   .IsApplicable(context.get(), transformation_context),
+               "The instruction may not be an atomic or barrier instruction. \
+                The operands position may be not equal 0 or 1. \
+                The position may be equal to one and the expected is zero.");
+
```

Comment:
Since you have assert death tests, you should add a test that passes a non-atomic and non-barrier opcode. You will get a different assertion message (if you make the other changes I suggested).

Reply:
Done.